xbmc/utils/URIUtils.cpp

```diff
@@ -199,9 +199,17 @@ std::string URIUtils::GetFileName(const std::string& strFileNameAndPath)
     return GetFileName(url.GetFileName());
   }
 
-  /* find the last slash */
-  const size_t slash = strFileNameAndPath.find_last_of("/\\");
-  return strFileNameAndPath.substr(slash+1);
+  int i = strFileNameAndPath.size() - 1;
+  while (i >= 0)
+  {
+    const char& ch = strFileNameAndPath[i];
```

Comment:
Can you do

```c++
const char ch = strFileNameAndPath[i];
```

? 

We don't need dangling lifetime and optimizers like this better.